File: src/main/java/com/koddy/server/member/application/usecase/CompleteProfileUseCase.java
package com.koddy.server.member.application.usecase;

import com.koddy.server.global.annotation.KoddyWritableTransactional;
import com.koddy.server.global.annotation.UseCase;
import com.koddy.server.member.application.usecase.command.CompleteMenteeProfileCommand;
import com.koddy.server.member.application.usecase.command.CompleteMentorProfileCommand;
import com.koddy.server.member.domain.model.mentee.Mentee;
import com.koddy.server.member.domain.model.mentor.Mentor;
import com.koddy.server.member.domain.repository.MenteeRepository;
import com.koddy.server.member.domain.repository.MentorRepository;

@UseCase
public class CompleteProfileUseCase {
    private final MentorRepository mentorRepository;
    private final MenteeRepository menteeRepository;

    public CompleteProfileUseCase(
            final MentorRepository mentorRepository,
            final MenteeRepository menteeRepository
    ) {
        this.mentorRepository = mentorRepository;
        this.menteeRepository = menteeRepository;
    }

    @KoddyWritableTransactional
    public void completeMentor(final CompleteMentorProfileCommand command) {
        final Mentor mentor = mentorRepository.getById(command.mentorId());
        mentor.completeInfo(
                command.introduction(),
                command.profileImageUrl(),
                command.mentoringPeriod(),
                command.timelines()
        );
    }

    @KoddyWritableTransactional
    public void completeMentee(final CompleteMenteeProfileCommand command) {
        final Mentee mentee = menteeRepository.getById(command.menteeId());
        mentee.completeInfo(command.introduction(), command.profileImageUrl());
    }
}

File: src/main/java/com/koddy/server/member/application/usecase/CompleteProfileUseCase.kt
package com.koddy.server.member.application.usecase

import com.koddy.server.global.annotation.KoddyWritableTransactional
import com.koddy.server.global.annotation.UseCase
import com.koddy.server.member.application.usecase.command.CompleteMenteeProfileCommand
import com.koddy.server.member.application.usecase.command.CompleteMentorProfileCommand
import com.koddy.server.member.domain.model.mentee.Mentee
import com.koddy.server.member.domain.model.mentor.Mentor
import com.koddy.server.member.domain.service.MemberReader

@UseCase
class CompleteProfileUseCase(
    private val memberReader: MemberReader,
) {
    @KoddyWritableTransactional
    fun completeMentor(command: CompleteMentorProfileCommand) {
        val mentor: Mentor = memberReader.getMentor(command.mentorId)
        mentor.completeProfile(
            command.introduction,
            command.profileImageUrl,
            command.mentoringPeriod,
            command.timelines,
        )
    }

    @KoddyWritableTransactional
    fun completeMentee(command: CompleteMenteeProfileCommand) {
        val mentee: Mentee = memberReader.getMentee(command.menteeId)
        mentee.completeProfile(
            command.introduction,
            command.profileImageUrl,
        )
    }
}
